src/Illuminate/Support/Arr.php

```diff
@@ -228,46 +228,51 @@ public static function flatten($array, $depth = INF)
     }
 
     /**
-     * Remove one or many array items from a given array using "dot" notation.
+     * Removes one item from a given array using "dot" notation".
      *
-     * @param  array  $array
-     * @param  array|string  $keys
+     * @param array $array
+     * @param string $key
+     * @param int $keyOffset
      * @return void
      */
-    public static function forget(&$array, $keys)
+    private static function forgetKey(&$array, &$key, $keyOffset = 0)
```

Comment:
What is the intend of `$keyOffset` as a parameter? This method is new and it is not being used as a parameter

Reply:
Do you mean intent? It's being used in the method to keep track of the position within $key

Reply:
Now I see that you implemented this as a recursive function, is there a performant way to avoid recursion?